pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1580,6 +1593,106 @@ func (vs *vSphereVMProvider) vmCreateDoPlacement(
 		return err
 	}
 
+	return processPlacementResult(vmCtx, vcClient, createArgs, *result)
+}
+
+// vmCreateDoPlacementByGroup places the VM from the group's placement result.
+func (vs *vSphereVMProvider) vmCreateDoPlacementByGroup(
+	vmCtx pkgctx.VirtualMachineContext,
+	vcClient *vcclient.Client,
+	createArgs *VMCreateArgs) error {
+
+	result := placement.DoesVMNeedPlacement(vmCtx)
```

Comment:
I don't think my intention is to ever make this public. And this seems out out of place to me - if the VM is part of a group, we should never short circuit it and make it wait for it entry in the group Status to be there.